opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -218,7 +227,12 @@ def record_batch(
     ) -> None:
         """See `opentelemetry.metrics.Meter.record_batch`."""
         for metric, value in record_tuples:
-            metric.get_handle(label_values).update(value)
+            if isinstance(metric, Counter):
```

Comment:
Is `record_batch` intended to be used with all kinds of metrics?
According to my understanding this is only related to `Measure`:
- https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#recordbatch-calling-convention
- https://github.com/open-telemetry/opentelemetry-go/blob/1191a07836765fa53af3ae1264ec42992f9d872f/api/metric/api.go#L82
- https://github.com/open-telemetry/opentelemetry-java/blob/master/api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java

Reply:
If you take a look at the first link, a `measurement` can be created from any metric instrument type (gauge, counter, measure), which are used in `record_batch`. `measurement` is not a first-level API-concept, merely a wrapper class that represents an instrument and value pair. See: https://github.com/open-telemetry/opentelemetry-specification/pull/250#discussion_r327880161

Reply:
You're right, for some reason I was thinking that `Measure` and `Measurement` were the same thing.

Reply:
Maybe the reason is that these names are very confusingly similar? 😄